packages/react-components/src/components/UserGuide/UserGuide.tsx

```diff
@@ -0,0 +1,247 @@
+  isVisible = false,
+  elementStyles,
+}) => {
+  const [parentElement, setParentElement] = useState<Element | null>(null);
+  const [rect, setRect] = useState<DOMRect | null>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const { refs, x, y, strategy, placement, update } = useFloating({
+    middleware: [shift(), flip()],
+    placement: cursorPosition,
+    open: true,
+    whileElementsMounted: autoUpdate,
+  });
+
+  const handleViewportChange = () => {
+    if (parentElement) {
+      setRect(
+        virtualReference(parentElement).getBoundingClientRect() as DOMRect
+      );
+    }
+  };
+
+  useEffect(() => {
+    if (parentElement !== null) {
+      window.addEventListener('resize', handleViewportChange);
+      window.addEventListener('scroll', handleViewportChange);
+
+      return () => {
+        window.removeEventListener('resize', handleViewportChange);
+        window.removeEventListener('scroll', handleViewportChange);
+      };
+    }
+  }, [parentElement]);
+
+  useEffect(() => {
+    const handleResize = () => update();
+    window.addEventListener('resize', handleResize);
+
+    return () => window.removeEventListener('resize', handleResize);
+  }, [update]);
+
+  useEffect(() => {
+    if (parentElementName) {
+      const element = document.querySelector(parentElementName);
+      setParentElement(element);
+      refs.setReference(element);
+    }
+  }, [parentElementName]);
+
+  useEffect(() => {
+    parentElement &&
+      setRect(
+        virtualReference(parentElement).getBoundingClientRect() as DOMRect
+      );
+  }, [parentElement]);
+
+  useEffect(() => {
+    if (parentElement && containerRef.current) {
+      containerRef.current.innerHTML = '';
+      const clonedElement = parentElement.cloneNode(true) as HTMLElement;
+      Object.assign(clonedElement.style, elementStyles);
+      containerRef.current.appendChild(clonedElement);
+    }
+
+    return () => {
+      if (containerRef.current) {
+        containerRef.current.innerHTML = '';
+      }
+    };
+  }, [parentElement, containerRef.current, isVisible]);
+
+  useEffect(() => {
+    if (isVisible) {
+      document.body.classList.add('user-guide-visible');
+    } else {
+      document.body.classList.remove('user-guide-visible');
+    }
+
+    return () => {
+      document.body.classList.remove('user-guide-visible');
+    };
+  }, [isVisible]);
+
+  const cloneReferenceElement = () => {
+    if (!rect) return null;
+
+    return (
+      <div
+        ref={containerRef}
+        style={{
+          position: 'absolute',
+          top: rect.top,
+          left: rect.left,
+          width: rect.width,
+          height: rect.height,
+          pointerEvents: 'none',
+        }}
+      />
+    );
+  };
+
+  return parentElement ? (
+    <>
+      <div
+        className={cx(styles[`${baseClass}__overlay`])}
+        style={{
+          display: isVisible ? 'block' : 'none',
```

Comment:
Can we add an option for a custom z-index on the overlay? Sometimes some elements have bigger zindex than overlay itself. I assume that the same index will be used for cloned element